Source/DafnyCore/Resolver/PreType/PreType.cs

```diff
@@ -0,0 +1,326 @@
+//-----------------------------------------------------------------------------
+//
+// Copyright by the contributors to the Dafny Project
+// SPDX-License-Identifier: MIT
+//
+//-----------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using Bpl = Microsoft.Boogie;
+
+namespace Microsoft.Dafny {
+  public abstract class PreType {
+    public PreType Normalize() {
```

Comment:
```suggestion

    public PreType Normalize() {
```

Reply:
It seems to me that for the existing `Type`, the function `Normalize` is defined within a class `Type` in `Types.cs`. Why are taking a different approach for `PreType`?

Reply:
A comment with a description of `Normalize` would be good to have. For `PreType`, why is there no need for e.g. `NormalizeExpand`?

Reply:
I added (the suggested blank line and) a description of the method.

For `Type`, there are three operations:
* `Normalize()` follows type proxies to get to the type for which it is a proxy.
* `NormalizeExpandKeepConstraints()` additionally follows any type synonym to the type it stands for.
* `NormalizeExpand()` additionally follows a subset type to its base type.

For `PreType`, the `Normalize()` method is analogous. Pre-types don't keep track of type synonyms or subset types; any such types given in the input program are `Type.NormalizeExpand()`ed before as part of turning it into a pre-type. Therefore, there is no analogous `PreType` form of `NormalizeExpandKeepConstraints()` and `NormalizeExpand()`.

I added this to the comment above `PreType.Normalize()`.